src/agent/rustjail/src/seccomp.rs

```diff
@@ -0,0 +1,218 @@
+    let mut filter = ScmpFilterContext::new_filter(def_action)?;
+
+    // Add extra architectures
+    for arch in &scmp.architectures {
+        let scmp_arch = ScmpArch::from_str(arch)?;
+        filter.add_arch(scmp_arch)?;
+    }
+
+    // Unset no new privileges bit
+    filter.set_no_new_privs_bit(false)?;
+
+    // Add a rule for each system call
+    for syscall in &scmp.syscalls {
+        if syscall.names.is_empty() {
+            return Err(anyhow!("syscall name is required"));
+        }
+
+        let action = ScmpAction::from_str(&syscall.action, Some(syscall.errno_ret))?;
+        if action == def_action {
+            continue;
+        }
+
+        for name in &syscall.names {
+            let syscall_num = get_syscall_from_name(name, None)?;
+
+            if syscall.args.is_empty() {
+                filter.add_rule(action, syscall_num, None)?;
+            } else {
+                let mut cmps: Vec<ScmpArgCompare> = Vec::new();
+
+                for arg in &syscall.args {
+                    if arg.op.is_empty() {
+                        return Err(anyhow!("seccomp opreator is required"));
+                    }
+
+                    let arg_cmp = ScmpArgCompare::new(
+                        arg.index,
+                        ScmpCompareOp::from_str(&arg.op)?,
+                        arg.value,
+                        Some(arg.value_two),
+                    );
+
+                    cmps.push(arg_cmp);
+                }
+
+                filter.add_rule(action, syscall_num, Some(&cmps))?;
+            }
+        }
+    }
+
+    // Set filtter attributes for each seccomp flag
+    for flag in &scmp.flags {
+        let scmp_attr = get_filter_attr_from_flag(flag)?;
+        filter.set_filter_attr(scmp_attr, 1)?;
+    }
+
+    // Load the filter
+    filter.load()?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::skip_if_not_root;
+    use libc::{dup2, process_vm_readv, EPERM};
+    use std::io::Error;
+    use std::ptr::null;
+
+    macro_rules! syscall_assert {
+        ($e1: expr, $e2: expr) => {
+            let mut errno: i32 = 0;
+            if $e1 < 0 {
+                errno = -Error::last_os_error().raw_os_error().unwrap();
+            }
+            assert_eq!(errno, $e2);
+        };
+    }
+
+    #[test]
+    fn test_get_filter_attr_from_flag() {
+        skip_if_not_root!();
+
+        assert_eq!(
+            get_filter_attr_from_flag("SECCOMP_FILTER_FLAG_TSYNC").unwrap(),
+            ScmpFilterAttr::CtlTsync
+        );
+
+        assert_eq!(get_filter_attr_from_flag("ERROR").is_err(), true);
+    }
+
+    #[test]
+    fn test_init_seccomp() {
+        skip_if_not_root!();
+
+        let data = r#"{
+            "defaultAction": "SCMP_ACT_ALLOW",
+            "architectures": [
+                "SCMP_ARCH_X86",
+                "SCMP_ARCH_X32",
+                "SCMP_ARCH_X86_64",
+                "SCMP_ARCH_AARCH64",
+                "SCMP_ARCH_ARM"
```

Comment:
Kata also works on PPC64 and s390x so won't this test fail unless you add them here?

Reply:
I added `PPC64` and `S390X` in the unit test.
To avoid endian mismatch error of `libseccomp`, I divided the architecture by endian.
Ref: https://github.com/seccomp/libseccomp/pull/138/commits/2745d8ac6b7d39051bccc8fd739808acc2741b57